main-unix.c

```diff
@@ -308,13 +342,19 @@ main (int argc,
       else
       {
         JERRY_ERROR_MSG ("Error: wrong format of the arguments\n");
+        print_usage ();
         return JERRY_STANDALONE_EXIT_CODE_FAIL;
       }
     }
     else if (!strcmp ("--abort-on-fail", argv[i]))
     {
       flags |= JERRY_FLAG_ABORT_ON_FAIL;
     }
+    else if (!strncmp ("-", argv[i], 1))
+    {
+      print_usage ();
```

Comment:
It seems that you have missed `--log-level`. At least, `print_usage ();` is not inserted there after the error message.

Moreover, I'm a bit concerned about printing the whole long usage info whenever something is wrong. Well-established tools normally just print a short hint in such cases, e.g.,
```
$ grep --XXX
grep: unrecognized option '--XXX'
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
```

(Help messages of such tools may also give ideas how to fine-tune the usage string to become more alike to what users are familiar with -- see my "unorthodoxy" comment above.)